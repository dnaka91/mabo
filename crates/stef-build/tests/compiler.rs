use std::{
    fs,
    path::{Path, PathBuf},
};

use insta::{assert_snapshot, glob, with_settings};
use stef_build::Opts;
use stef_parser::Schema;

fn strip_path(path: &Path) -> PathBuf {
    path.strip_prefix(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/inputs"))
        .or_else(|_| path.strip_prefix(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/inputs_extra")))
        .unwrap()
        .to_owned()
}

#[test]
fn compile_schema() {
    glob!("inputs/*.stef", |path| {
        let input = fs::read_to_string(path).unwrap();
        let value = Schema::parse(input.as_str(), Some(&strip_path(path))).unwrap();
        let value = stef_compiler::simplify_schema(&value);
        let value = stef_build::compile_schema(&Opts::default(), &value);
        let value = prettyplease::unparse(&syn::parse2(value.clone()).unwrap());

        with_settings!({
            description => input.trim(),
            omit_expression => true,
        }, {
            assert_snapshot!("compile", value);
        });
    });
}

#[test]
fn compile_schema_extra() {
    glob!("inputs_extra/*.stef", |path| {
        let input = fs::read_to_string(path).unwrap();
        let value = Schema::parse(input.as_str(), Some(&strip_path(path))).unwrap();
        let value = stef_compiler::simplify_schema(&value);
        let value = stef_build::compile_schema(&Opts::default(), &value);
        let value = prettyplease::unparse(&syn::parse2(value.clone()).unwrap());

        with_settings!({
            description => input.trim(),
            omit_expression => true,
        }, {
            assert_snapshot!("compile_extra", value);
        });
    });
}
