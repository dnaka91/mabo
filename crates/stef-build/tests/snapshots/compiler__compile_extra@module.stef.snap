---
source: crates/stef-build/tests/compiler.rs
expression: "/// Hello world!\nmod sample {}"
input_file: crates/stef-build/tests/inputs_extra/module.stef
---
#[allow(unused_imports)]
use ::stef::buf::{Decode, Encode};
/// Hello world!
pub mod sample {
    #[allow(unused_imports)]
    use ::stef::buf::{Decode, Encode};
}

